etc/nginx/nginx-ssl.conf.ctmpl

```diff
@@ -0,0 +1,86 @@
+# This is an example Nginx configuration template file.
+# Adjust the values below as required for your application.
+
+user  nginx;
+worker_processes  1;
+
+error_log  /var/log/nginx/error.log warn;
+pid        /var/run/nginx.pid;
+
+events {
+    worker_connections  1024;
+}
+
+
+http {
+    include       /etc/nginx/mime.types;
+    default_type  application/octet-stream;
+
+    map $status $isError {
+        ~^2 0;
+        default 1;
+    }
+
+    log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
+                      '$status $body_bytes_sent "$http_referer" '
+                      '"$http_user_agent" "$http_x_forwarded_for"';
+
+    access_log  /var/log/nginx/access.log  main;
+
+    sendfile        on;
+    #tcp_nopush     on;
+
+    keepalive_timeout  65;
+
+    #gzip  on;
+
+    {{ $backend := env "BACKEND" }}
+    {{ $acme_domain := env "ACME_DOMAIN" }}
+    {{ if service $backend }}
+    upstream {{ $backend }} {
+        # write the address:port pairs for each healthy backend instance
+        {{range service $backend }}
+        server {{.Address}}:{{.Port}};
+        {{end}}
+        least_conn;
+    }{{ end }}
+
+    server {
+        listen       80;
+        server_name  _;
+
+        location /nginx-health {
+            stub_status;
+            allow 127.0.0.1;
+            deny all;
+            access_log /var/log/nginx/access.log  main if=$isError;
+        }
+
+        {{ if $acme_domain }}
+        location /.well-known/acme-challenge {
+            alias /var/www/acme/challenge;
+        }
+        {{ end }}
+        listen 443 ssl;
```

Comment:
Should these be under separate server blocks? With this setup we're accepting requests for the default (`_`) hostname on port 80 and our secured domain on port 443 together. I think we want to have the port 80 server block only serve the items we absolutely need it to and then redirect to the port 443 server block for everything else.